Script Issue

Hi,

Not sure that the script is working correctly. Log output below:

##### Current OpenVPN Version:
OpenVPN 2.3.2 mips-unknown-linux-gnu [SSL (OpenSSL)] [LZO] [EPOLL] [PKCS11] [eurephia] [MH] [IPv6] built on Dec  5 2014
Originally developed by James Yonan
Copyright (C) 2002-2010 OpenVPN Technologies, Inc. <sales@openvpn.net>
Compile time defines: enable_crypto=yes enable_debug=yes enable_def_auth=yes enable_dependency_tracking=no enable_dlopen=unknown enable_dlopen_self=unknown enable_dlopen_self_static=unknown enable_eurephia=yes enable_fast_install=yes enable_fragment=yes enable_http_proxy=yes enable_iproute2=yes enable_libtool_lock=yes enable_lzo=yes enable_lzo_stub=no enable_maintainer_mode=no enable_management=yes enable_multi=yes enable_multihome=yes enable_pam_dlopen=no enable_password_save=yes enable_pedantic=no enable_pf=yes enable_pkcs11=yes enable_plugin_auth_pam=yes enable_plugin_down_root=yes enable_plugins=yes enable_port_share=yes enable_selinux=no enable_server=yes enable_shared=yes enable_shared_with_static_runtimes=no enable_small=no enable_socks=yes enable_ssl=yes enable_static=yes enable_strict=no enable_strict_options=no enable_systemd=no enable_win32_dll=yes enable_x509_alt_username=yes with_crypto_library=openssl with_gnu_ld=yes with_ifconfig_path=/sbin/ifconfig with_iproute_path=/sbin/ip with_mem_check=no with_plugindir='${prefix}/lib/openvpn' with_route_path=/sbin/route with_sysroot=no
##### Checking if the debian repository already exists in /etc/apt/sources.list ...
##### Looking for 'deb http://archive.debian.org/debian/ wheezy main non-free contrib'
##### Source already added to the list.
##### Gathering required binaries in order to compile OpenVPN 2.4.12
##### Running 'apt-get update'
Hit http://archive.debian.org wheezy Release.gpg
Hit http://archive.debian.org wheezy Release
Hit http://archive.debian.org wheezy/contrib Translation-en
Hit http://archive.debian.org wheezy/main Translation-en
Hit http://archive.debian.org wheezy/non-free Translation-en
Hit http://archive.debian.org wheezy/main mips Packages
Hit http://archive.debian.org wheezy/non-free mips Packages
Hit http://archive.debian.org wheezy/contrib mips Packages
Reading package lists... Done
##### Running 'apt-get install wget -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
wget is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
##### Running 'apt-get install apt-utils -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
apt-utils is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
##### Running 'apt-get install libssl-doc -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
libssl-doc is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
##### Running 'apt-get install libc6/wheezy -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc6'
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc-bin' because of 'libc6'
The following extra packages will be installed:
  libc-bin
Suggested packages:
  glibc-doc locales
The following packages will be DOWNGRADED:
  libc-bin libc6
0 upgraded, 0 newly installed, 2 downgraded, 0 to remove and 1 not upgraded.
Need to get 5696 kB of archives.
After this operation, 77.8 kB of additional disk space will be used.
E: There are problems and -y was used without --force-yes
##### Running 'apt-get install libc6-dev/wheezy -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc6-dev'
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc6' because of 'libc6-dev'
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc-bin' because of 'libc6'
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc-dev-bin' because of 'libc6-dev'
The following extra packages will be installed:
  cpp cpp-4.6 gcc gcc-4.6 gcc-4.6-base libc-bin libc-dev-bin libc6 libgomp1
  libmpc2 libmpfr4 linux-libc-dev manpages manpages-dev
Suggested packages:
  cpp-doc gcc-4.6-locales gcc-multilib make autoconf automake1.9 libtool flex
  bison gdb gcc-doc gcc-4.6-multilib libmudflap0-4.6-dev gcc-4.6-doc
  libgcc1-dbg libgomp1-dbg libquadmath-dbg libmudflap0-dbg glibc-doc
  man-browser
The following NEW packages will be installed:
  cpp cpp-4.6 gcc gcc-4.6 gcc-4.6-base libc-dev-bin libc6-dev libgomp1 libmpc2
  libmpfr4 linux-libc-dev manpages manpages-dev
The following packages will be DOWNGRADED:
  libc-bin libc6
0 upgraded, 13 newly installed, 2 downgraded, 0 to remove and 1 not upgraded.
Need to get 20.6 MB of archives.
After this operation, 37.2 MB of additional disk space will be used.
E: There are problems and -y was used without --force-yes
##### Running 'apt-get install libc-dev-bin/wheezy -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc-dev-bin'
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc6' because of 'libc-dev-bin'
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc-bin' because of 'libc6'
The following extra packages will be installed:
  libc-bin libc6 manpages manpages-dev
Suggested packages:
  glibc-doc locales man-browser
The following NEW packages will be installed:
  libc-dev-bin manpages manpages-dev
The following packages will be DOWNGRADED:
  libc-bin libc6
0 upgraded, 3 newly installed, 2 downgraded, 0 to remove and 1 not upgraded.
Need to get 8455 kB of archives.
After this operation, 3088 kB of additional disk space will be used.
E: There are problems and -y was used without --force-yes
##### Running 'apt-get install libc-bin/wheezy -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [mips]) for 'libc-bin'
The following packages will be DOWNGRADED:
  libc-bin
0 upgraded, 0 newly installed, 1 downgraded, 0 to remove and 1 not upgraded.
Need to get 1443 kB of archives.
After this operation, 76.8 kB of additional disk space will be used.
E: There are problems and -y was used without --force-yes
##### Running 'apt-get install locales/wheezy -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
Selected version '2.13-38+deb7u10' (Debian:7.11/oldoldstable [all]) for 'locales'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 locales : Depends: glibc-2.13-1
           Depends: debconf (>= 0.5) but it is not going to be installed or
                    debconf-2.0
E: Unable to correct problems, you have held broken packages.
##### Running 'apt-get install sudo/wheezy -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
sudo is already the newest version.
Selected version '1.8.5p2-1+nmu3+deb7u1' (Debian:7.11/oldoldstable [mips]) for 'sudo'
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
##### Running 'apt-get install build-essential -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 build-essential : Depends: libc6-dev but it is not going to be installed or
                            libc-dev
                   Depends: g++ (>= 4:4.4.3) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
##### Running 'apt-get install aptitude -y'
Reading package lists... Done
Building dependency tree
Reading state information... Done
aptitude is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
##### Apt-get Commands Complete
##### Running aptitude installation tasks
##### Running 'printf '%s\n' n 2 Y Y |aptitude install libssl-dev'
UpgradeOpenVPN.sh: line 66: 22742 Done                    printf '%s\n' n 2 Y Y
     22743 Segmentation fault      | aptitude install libssl-dev
##### Running 'aptitude install liblzo2-dev -y'
UpgradeOpenVPN.sh: line 66: 22744 Segmentation fault      aptitude install liblzo2-dev -y
##### Running 'aptitude install libpam0g-dev -y'
UpgradeOpenVPN.sh: line 66: 22745 Segmentation fault      aptitude install libpam0g-dev -y
##### Running 'aptitude install libtool -y'
UpgradeOpenVPN.sh: line 66: 22746 Segmentation fault      aptitude install libtool -y
##### Getting OpenVPN 2.4.12 from openvpn.org...
--2022-08-26 23:19:47--  https://swupdate.openvpn.org/community/releases/openvpn-2.4.12.tar.gz
Resolving swupdate.openvpn.org (swupdate.openvpn.org)... 104.16.183.48, 104.16.184.48
Connecting to swupdate.openvpn.org (swupdate.openvpn.org)|104.16.183.48|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1480810 (1.4M) [application/x-tar]
Saving to: `openvpn-2.4.12.tar.gz.2'

100%[======================================>] 1,480,810   2.66M/s   in 0.5s

2022-08-26 23:19:51 (2.66 MB/s) - `openvpn-2.4.12.tar.gz.2' saved [1480810/1480810]

##### Extracting OpenVPN 2.4.1 tarball...
##### Configuring OpenVPN options...
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... no
checking whether make supports nested variables... no
checking build system type... ./config.guess: unable to guess system type

NOTE: MIPS GNU/Linux systems require a C compiler to fully recognize
the system type. Please install a C compiler and try again.

This script (version 2018-08-29), has failed to recognize the
operating system you are using. If your script is old, overwrite *all*
copies of config.guess and config.sub with the latest versions from:

  https://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess
and
  https://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub

If ./config.guess has already been updated, send the following data and any
information you think might be pertinent to config-patches@gnu.org to
provide the necessary information to handle your system.

config.guess timestamp = 2018-08-29

uname -m = mips64
uname -r = 3.10.107-UBNT
uname -s = Linux
uname -v = #1 SMP Wed Oct 20 08:38:35 UTC 2021

/usr/bin/uname -p =
/bin/uname -X     =

hostinfo               =
/bin/universe          =
/usr/bin/arch -k       =
/bin/arch              =
/usr/bin/oslevel       =
/usr/convex/getsysinfo =

UNAME_MACHINE = "mips64"
UNAME_RELEASE = "3.10.107-UBNT"
UNAME_SYSTEM  = "Linux"
UNAME_VERSION = "#1 SMP Wed Oct 20 08:38:35 UTC 2021"
configure: error: cannot guess build type; you must specify one
##### Make...
UpgradeOpenVPN.sh: line 83: make: command not found
##### Make install...
UpgradeOpenVPN.sh: line 86: make: command not found
##### Dumping new openvpn version:
OpenVPN 2.3.2 mips-unknown-linux-gnu [SSL (OpenSSL)] [LZO] [EPOLL] [PKCS11] [eurephia] [MH] [IPv6] built on Dec  5 2014
Originally developed by James Yonan
Copyright (C) 2002-2010 OpenVPN Technologies, Inc. <sales@openvpn.net>
Compile time defines: enable_crypto=yes enable_debug=yes enable_def_auth=yes enable_dependency_tracking=no enable_dlopen=unknown enable_dlopen_self=unknown enable_dlopen_self_static=unknown enable_eurephia=yes enable_fast_install=yes enable_fragment=yes enable_http_proxy=yes enable_iproute2=yes enable_libtool_lock=yes enable_lzo=yes enable_lzo_stub=no enable_maintainer_mode=no enable_management=yes enable_multi=yes enable_multihome=yes enable_pam_dlopen=no enable_password_save=yes enable_pedantic=no enable_pf=yes enable_pkcs11=yes enable_plugin_auth_pam=yes enable_plugin_down_root=yes enable_plugins=yes enable_port_share=yes enable_selinux=no enable_server=yes enable_shared=yes enable_shared_with_static_runtimes=no enable_small=no enable_socks=yes enable_ssl=yes enable_static=yes enable_strict=no enable_strict_options=no enable_systemd=no enable_win32_dll=yes enable_x509_alt_username=yes with_crypto_library=openssl with_gnu_ld=yes with_ifconfig_path=/sbin/ifconfig with_iproute_path=/sbin/ip with_mem_check=no with_plugindir='${prefix}/lib/openvpn' with_route_path=/sbin/route with_sysroot=no

--- a/USG/UpgradeOpenVPNClient/UpgradeOpenVPN.sh
+++ b/USG/UpgradeOpenVPNClient/UpgradeOpenVPN.sh
@@ -3,8 +3,8 @@ if [ "$EUID" -ne 0 ]
     exit
 fi
 
-if openvpn --version | grep -q "OpenVPN 2.4.12";
-    then echo "OpenVPN is already on version 2.4.12"
+if openvpn --version | grep -q "OpenVPN 2.5.5";
+    then echo "OpenVPN is already on version 2.5.5"
     exit
 fi
 
@@ -29,7 +29,7 @@ fi
 
 # Next we define the set of things that we need to install using apt-get
 
-echo "$outputTag Gathering required binaries in order to compile OpenVPN 2.4.12"
+echo "$outputTag Gathering required binaries in order to compile OpenVPN 2.5.5"
 
 aptGetCmds=("apt-get update"
 "apt-get install wget -y"
@@ -42,7 +42,9 @@ aptGetCmds=("apt-get update"
 "apt-get install locales/wheezy -y"
 "apt-get install sudo/wheezy -y"
 "apt-get install build-essential -y"
-"apt-get install aptitude -y")
+"apt-get install liblzo2-dev -y"
+"apt-get install libpam0g-dev -y"
+)
 
 for i in ${!aptGetCmds[@]}; do
     echo "$outputTag Running '${aptGetCmds[$i]}'"
@@ -51,33 +53,43 @@ done
 
 echo "$outputTag Apt-get Commands Complete"
 
-# The printf statements automate the prompt responses.
-aptitudeCmds=(
-    # Automate responses: no (reject proposed solution), 2 (option 2 - install bin86), remaining prompts "Yes"
-    "printf '%s\n' n 2 Y Y |aptitude install libssl-dev"
-    "aptitude install liblzo2-dev -y"
-    "aptitude install libpam0g-dev -y"
-    "aptitude install libtool -y")
-
-echo "$outputTag Running aptitude installation tasks"
-
-for i in ${!aptitudeCmds[@]}; do
-    echo "$outputTag Running '${aptitudeCmds[$i]}'"
-    eval ${aptitudeCmds[$i]}
-done
-
 cd /tmp
 
-echo "$outputTag Getting OpenVPN 2.4.12 from openvpn.org..."
-wget https://swupdate.openvpn.org/community/releases/openvpn-2.4.12.tar.gz
+echo "$outputTag ########################################"
 
-echo "$outputTag Extracting OpenVPN 2.4.1 tarball..."
-tar -xzf openvpn-2.4.12.tar.gz
+echo "$outputTag Getting libssl 1.1.1m from openssl.org..."
+wget https://www.openssl.org/source/openssl-1.1.1m.tar.gz
 
-cd openvpn-2.4.12/
+echo "$outputTag Extracting libssl 1.1.1m tarball..."
+tar -xzf openssl-1.1.1m.tar.gz
+cd openssl-1.1.1m/
+
+echo "$outputTag Configuring libssl options..."
+./Configure --prefix=/usr linux-mips32
+
+echo "$outputTag Make..."
+make
+
+echo "$outputTag Make test..."
+make test
+
+# TODO: Should check make output and fail install if tests fails. There probably is some return code to check.
+
+echo "$outputTag Make install..."
+make install
+
+echo "$outputTag ########################################"
+
+echo "$outputTag Getting OpenVPN 2.5.5 from openvpn.org..."
+wget https://swupdate.openvpn.org/community/releases/openvpn-2.5.5.tar.gz
+
+echo "$outputTag Extracting OpenVPN 2.5.5 tarball..."
+tar -xzf openvpn-2.5.5.tar.gz
+
+cd openvpn-2.5.5/
 
 echo "$outputTag Configuring OpenVPN options..."
-./configure --prefix=/usr
+./configure --prefix=/usr --disable-plugins
 
 echo "$outputTag Make..."
 make

